tests/storage/hotplug.go

```diff
@@ -1019,10 +1026,91 @@ var _ = SIGDescribe("Hotplug", func() {
 				addVolumeFunc(vmi.Name, vmi.Namespace, volumeName, dv.Name, v1.DiskBusSCSI, false, "")
 				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				VerifyVolumeAndDiskVMIAdded(virtClient, vmi, volumeName)
+				verifyVolumeAndDiskVMIAdded(virtClient, vmi, volumeName)
 				verifyVolumeStatus(vmi, v1.VolumeReady, "", volumeName)
 			})
 		})
+
+		Context("disk mutating sidecar", func() {
+			const (
+				hookSidecarImage     = "example-disk-mutation-hook-sidecar"
+				sidecarContainerName = "hook-sidecar-0"
+				newDiskImgName       = "kubevirt-disk.img"
+			)
+
+			var (
+				vm *v1.VirtualMachine
+				dv *cdiv1.DataVolume
+				sc string
+			)
+
+			renderSidecar := func() map[string]string {
+				return map[string]string{
+					"hooks.kubevirt.io/hookSidecars":         fmt.Sprintf(`[{"image": "%s/%s:%s", "imagePullPolicy": "IfNotPresent"}]`, flags.KubeVirtUtilityRepoPrefix, hookSidecarImage, flags.KubeVirtUtilityVersionTag),
+					"diskimage.vm.kubevirt.io/bootImageName": newDiskImgName,
+				}
+			}
+
+			BeforeEach(func() {
+				if !libstorage.HasCDI() {
+					Skip("Skip tests when CDI is not present")
+				}
+				var exists bool
+				sc, exists = libstorage.GetRWXBlockStorageClass()
+				if !exists {
+					Skip("Skip test when RWXBlock storage class is not present")
+				}
+			})
+
+			AfterEach(func() {
+				if vm != nil {
+					err := virtClient.VirtualMachine(vm.Namespace).Delete(vm.Name, &metav1.DeleteOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					vm = nil
+				}
+				libstorage.DeleteDataVolume(&dv)
+			})
+
+			DescribeTable("should be able to add and remove volumes", func(addVolumeFunc addVolumeFunction, removeVolumeFunc removeVolumeFunction, volumeMode corev1.PersistentVolumeMode, vmiOnly bool) {
+				var err error
+				url := cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros)
+				dv = libstorage.NewDataVolumeWithRegistryImport(url, util.NamespaceTestDefault, k8sv1.ReadWriteMany)
+				tests.SetDataVolumeForceBindAnnotation(dv)
+
+				dv.Spec.PVC.Resources.Requests[k8sv1.ResourceStorage] = resource.MustParse("6Gi")
```

Comment:
The kubevirt ci cluster has ~30Gi of ceph storage available. Cirros doesn't need 6Gi, I would make this maybe 512Mi or 256Mi if possible. This prevents errors due to several tests running at the same time exhausting the storage space.

Reply:
Changed to 256Mi. Thanks for the suggestion.